### PR TITLE
debt/[CORE-1750]: Versioned Transaction Migration

### PR DESCRIPTION
**Pull request by iamalperen**
We were using `Transaction` but it's legacy and deprecated. We are not migrating our structure to `VersionedTransaction`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,9 @@
     "tsconfig-paths": "^4.1.0",
     "typescript": "^4.8.3"
   },
-  "dependencies": {},
+  "dependencies": {
+    "@solana/wallet-adapter-phantom": "^0.9.23"
```
Comment (iamalperen) [inline package.json:38]: This is for VersionedTransaction addition to solana web3 library. phantom adapter includes the VersionedTransaction

File: packages/blockchain-sdk-solana/src/wallet-adapter/dialect-solana-wallet-adapter.interface.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import type { PublicKey } from '@solana/web3.js';
 export interface DialectSolanaWalletAdapter {
   publicKey?: PublicKey;
   signTransaction?: SignerWalletAdapterProps['signTransaction'];
-  signAllTransactions?: SignerWalletAdapterProps['signAllTransactions'];
+  signAllTransactions: SignerWalletAdapterProps['signAllTransactions'];
```
Comment (tsmbl) [inline packages/blockchain-sdk-solana/src/wallet-adapter/dialect-solana-wallet-adapter.interface.ts:10]: why this field became mandatory?

**Review by tsmbl — COMMENTED** [review]
should we also migrate token sig/validation to versioned tx?

**Comment by iamalperen** [comment]
> should we also migrate token sig/validation to versioned tx?

I think so

**Comment by tsmbl** [comment]
produces comp errors for me

should we update @solana/wallet-adapter-base version to actualize `SignerWalletAdapterProps`?

File: packages/blockchain-sdk-solana/src/wallet-adapter/node-dialect-solana-wallet-adapter.ts [hunk 3]
```diff
@@ -38,16 +44,22 @@ export class NodeDialectSolanaWalletAdapter
...
-    return tx;
+  async signTransaction<T extends Transaction | VersionedTransaction>(
+    transaction: T,
+  ): Promise<T> {
+    const signer: Signer = {
+      publicKey: this.keypair.publicKey,
+      secretKey: this.keypair.secretKey,
+    };
+    const signers: Signer[] = [signer];
+    (transaction as VersionedTransaction).sign(signers);
```
Comment (tsmbl) [inline packages/blockchain-sdk-solana/src/wallet-adapter/node-dialect-solana-wallet-adapter.ts:55]: unsafe cast

**Review by tsmbl — CHANGES_REQUESTED** [review]

File: packages/blockchain-sdk-solana/src/wallet-adapter/node-dialect-solana-wallet-adapter.ts (hunk 3 above)
Comment (iamalperen) [reply packages/blockchain-sdk-solana/src/wallet-adapter/node-dialect-solana-wallet-adapter.ts:55]: Why do you think it's unsafe? You can see the definition from the function, there are type types available and here I am trying to make sure about its type. What is your suggestion?

Comment (tsmbl) [reply packages/blockchain-sdk-solana/src/wallet-adapter/node-dialect-solana-wallet-adapter.ts:55]: does code compile w/o cast?

**Comment by zhelezkov** [comment]
just curios, is sdk migration is necessary or we can avoid it?

**Review by kevindingens — APPROVED** [review]

**Review by tsmbl — APPROVED** [review]

File: packages/blockchain-sdk-solana/src/wallet-adapter/dialect-solana-wallet-adapter-wrapper.ts [hunk 4]
```diff
@@ -1,4 +1,8 @@
-import type { PublicKey, Transaction } from '@solana/web3.js';
+import type {
+  PublicKey,
+  Transaction,
+  VersionedTransaction,
+} from '@solana/web3.js';
```
Comment (dshuvalov) [inline packages/blockchain-sdk-solana/src/wallet-adapter/dialect-solana-wallet-adapter-wrapper.ts:5]: Should we update this pacakge to the latest?

**Review by dshuvalov — CHANGES_REQUESTED** [review]
Why we have diff in yarn.lock file but package.json the same?

**Review by dshuvalov — CHANGES_REQUESTED** [review]
Few question

1. `@solana/web3.js` uses as `peerDependecy` and `devDep` in the most of the packages. Why you didn't update all of them?
2. Why did you update the version of the package in the `sdk` when you made changes in the blockchain-sdk-solana package?
3. I would suggest to bump version up to `^1.54.0` in peerDeps of blockchains-sdk-solana package. Because `VersionedTransaction` was introduced only at `1.54.0` version https://github.com/solana-labs/solana-web3.js/releases/tag/v1.54.0

**Comment by tsmbl** [comment]
> As for me, switching from `Transaction` to `VersionedTransaction` is a breaking change and I am not sure that it is a backward compatible @tsmbl @zhelezkov @kevindingens wdyt?

why breaking? I think both will supported

**Comment by iamalperen** [comment]
> As for me, switching from `Transaction` to `VersionedTransaction` is a breaking change and I am not sure that it is a backward compatible @tsmbl @zhelezkov @kevindingens wdyt?

We are supporting both of them. Why do you think it as breaking?

**Comment by dshuvalov** [comment]
@iamalperen @tsmbl I checked the code and agree with you. It seems we support both versions. 
So, let's bump the peer deep as I mentioned at the comment above

**Pull request merged**